presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -3141,6 +3141,20 @@ public void testPartitionedTablesFailWithAvroSchemaUrl()
         assertQueryFails(createSql, "Bucketing/Partitioning columns not supported when Avro schema url is set");
     }
 
+    @Test
+    public void testPruneEmptyPartitionFailure()
+    {
+        // TODO: fix creating empty partitions for non-bucketed tables
```

Comment:
Theoretically, someone wanting to improve the test by removing bucketing, should also retest that the simplified test still would caught the bug.
Maybe, instead of TODO here, create a simpler table schema (without bucketing) and instead of `system.create_empty_partition` you would do `insert`.

Reply:
I figured out a different way to reproduce the bug, without needing an empty partition. But it has the downside that it's not guaranteed to work, as it relies on our current evaluation order. (though I suspect changing that would break a huge number of real queries)